siege_utilities/geo/providers/boundary_providers.py

```
@@ -0,0 +1,399 @@
+        url = _GADM_BASE_URL.format(
+            version=self._version,
+            country=country_code,
+            level=numeric_level,
+        )
+
+        try:
+            import geopandas as gpd
+        except ImportError:
+            raise ImportError(
+                'GADMProvider requires geopandas. Install with: pip install siege_utilities[geo]'
+            )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)